### PR TITLE
docs: update RELEASE.md

### LINKED ISSUE #12864: Document release process with the automation workflow
The current [RELEASES.md](https://github.com/elastic/apm-server/blob/main/dev_docs/RELEASES.md) doesn't mention how to use the CI automation or when to run them. This takes precious time to figure things out OR to create the manual PR. We should document the updated/streamlined process using the CI jobs.

### PR DESCRIPTION
**Pull request by endorama**
This PR aims at refreshing the release documentation for `apm-server`. I'll use it to collect changes as I go through the release process as release manager.

Closes #12864

### CODE REVIEW

File: dev_docs/RELEASES.md [hunk 1]
```diff
@@ -17,9 +17,9 @@ For patch releases, only the version on the existing major and minor version bra
   * Review existing [changelogs/head](https://github.com/elastic/apm-server/tree/main/changelogs/head.asciidoc) to ensure all relevant notes have been added.
```
Comment (endorama) [inline dev_docs/RELEASES.md:17]: Question: how do I make sure there are no PR to be merged? I checked all PRs that don't have a `backport-skip` label applied through [this search](https://github.com/elastic/apm-server/pulls?q=is%3Apr+sort%3Acreated-desc+-label%3Abackport-skip). If that is appropriate I would add it to the documentation to help streamline this step.

File: dev_docs/RELEASES.md [hunk 2]
```diff
@@ -17,9 +17,9 @@ For patch releases, only the version on the existing major and minor version bra
...
   * Move changelog entries from _head_ to _release_version_:
     * Minor version:
-      Create new changelog file from [changelogs/head.asciidoc](https://github.com/elastic/apm-server/blob/main/changelogs/head.asciidoc)
-      If changes should not be backported, keep them in the _changelogs/head.asciidoc_ file.
-      Don't forget to `include` and link to the new file in [main changelog](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc) and the [release notes](https://github.com/elastic/apm-server/blob/main/docs/release-notes.asciidoc) file. [(Sample PR)](https://github.com/elastic/apm-server/pull/7956/files)
+      Create new changelog file from [changelogs/head.asciidoc](https://github.com/elastic/apm-server/blob/main/changelogs/head.asciidoc).  
+      If changes should not be backported, keep them in the _changelogs/head.asciidoc_ file.
```
Comment (endorama) [inline dev_docs/RELEASES.md:21]: > If changes should not be backported

Question: backported from where to where? Clarify how to check this.

Comment (endorama) [reply dev_docs/RELEASES.md]: Line removed as this part is covered by automation.

File: dev_docs/RELEASES.md [hunk 3]
```diff
@@ -9,34 +9,18 @@ For patch releases, only the version on the existing major and minor version bra
...
-    Trigger a new beats update, once the beats branch is also created.
-    Remove the [changelogs/head.asciidoc](https://github.com/elastic/apm-server/blob/main/changelogs/head.asciidoc) file from the release branch.
-
-  * Main branch:
-    Update [.mergify.yml](https://github.com/elastic/apm-server/blob/main/.mergify.yml) with a new backport rule for the next version,
-    and update versions to next minor version, e.g. [#2804](https://github.com/elastic/apm-server/pull/2804).
-
-  The release manager will ping the teams, but you can already prepare this in advance on the day after Feature Freeze.
+* Update Changelog
+  * This step is automated
```
Comment (carsonip) [inline dev_docs/RELEASES.md:19]:
```suggestion
* Trigger release workflow manually
```
it does a lot more than "update changelog" as described below

File: dev_docs/RELEASES.md [hunk 4]
```diff
@@ -55,11 +39,12 @@ For patch releases, only the version on the existing major and minor version bra
...
-  * Add other test cases that require manual testing, such as test scenarios on ESS, that are not covered by automated tests or
-    OS compatibility smoke tests for supporting new operating systems.
+  Create a github issue for testing the release branch ([use the GitHub issue `test plan` template](https://github.com/elastic/apm-server/issues/new?assignees=&labels=test-plan&projects=&template=test-plan.md)), It should contain:
+  * A link to all PRs in the APM Server repository that need to be tested manually. Use the `test-plan` label and the version label (create it if it does not exist).
+    to create an overview over the PRs that need testing. For example, [test plan link for 8.3.0](https://github.com/elastic/apm-server/issues?q=label%3Atest-plan+is%3Aclosed+label%3Av8.3.0).  
+    What we aim for is testing all functional changes applied to the new version. Review any PR updating `elastic/go-docappener` and `elastic/apm-data` dependencies, as some functional changes happens through these dependencies.
```
Comment (carsonip) [inline dev_docs/RELEASES.md:45]:
```suggestion
    What we aim for is testing all functional changes applied to the new version. Review any PR updating `elastic/go-docappender` and `elastic/apm-data` dependencies, as some functional changes happens through these dependencies.  
```

File: dev_docs/RELEASES.md [hunk 5]
```diff
@@ -55,11 +39,12 @@ For patch releases, only the version on the existing major and minor version bra
 
 * Test plan
 
-  Create a github issue for testing the release branch (follow the GitHub issue template for the test plan), It should contain:
-  * A link to all PRs in the APM Server repository that need to be tested manually. Use the `test-plan*` labels and the version labels
-    to create an overview over the PRs that need testing. For example, [test plan link for 8.3.0](https://github.com/elastic/apm-server/issues?q=label%3Atest-plan+is%3Aclosed+label%3Av8.3.0).
-  * Add other test cases that require manual testing, such as test scenarios on ESS, that are not covered by automated tests or
-    OS compatibility smoke tests for supporting new operating systems.
+  Create a github issue for testing the release branch ([use the GitHub issue `test plan` template](https://github.com/elastic/apm-server/issues/new?assignees=&labels=test-plan&projects=&template=test-plan.md)), It should contain:
```
Comment (carsonip) [inline dev_docs/RELEASES.md:42]: What about including a recent example?

**Review by carsonip — COMMENTED** [review]
I think it is missing docs release process. It used to be done by docs team but it is now our responsibility to do so. Can you figure out at which stage that should happen and update the docs, and give an example?

File: dev_docs/RELEASES.md (hunk 5 above)
Comment (endorama) [reply dev_docs/RELEASES.md:41]: Are you referring to the link on line 41 or 43? This link brings you to the new issue view with the test-plan template selected (there is a separate PR for updating the test plan). On line 43 instead there is a link to a 8.3.0 test plan, which looks like what you would like to see updated?

**Comment by endorama** [comment]
@carsonip thanks for pointing to the doc release. I'll dig into that.

File: dev_docs/RELEASES.md (hunk 5 above)
Comment (carsonip) [reply dev_docs/RELEASES.md:41]: I'd like an new link to a recent test plan issue, in addition to the existing link to the issue view with filtered with test-plan label

File: dev_docs/RELEASES.md [hunk 6]
```diff
@@ -9,34 +9,17 @@ For patch releases, only the version on the existing major and minor version bra
 
 ## Feature Freeze
 
-* For patch releases, ensure all relevant backport PRs are merged.
-  We use backport labels on PRs and automation to ensure labels are set.
+* For **patch releases**:
+  * ensure all relevant backport PRs are merged. We use backport labels on PRs and automation to ensure labels are set.
```
Comment (bmorelli25) [inline dev_docs/RELEASES.md:14]:
```suggestion
  * Ensure all relevant backport PRs are merged. Backport PRs are automatically opened based on `backport-X.Y` labels added to PRs opened against `main`.
  * Add release notes
    * Review the [changelogs/head](https://github.com/elastic/apm-server/tree/main/changelogs/head.asciidoc) file and move all relevant release notes from `head.asciidoc` to `release_version.asciidoc.
      * Minor version:
        Create a new changelog file from [`changelogs/head.asciidoc`](https://github.com/elastic/apm-server/blob/main/changelogs/head.asciidoc)
        If changes do not apply to the version being released, keep them in the `changelogs/head.asciidoc` file.
        Don't forget to `include` and link to the new file in [main changelog](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc) file and the [release notes](https://github.com/elastic/apm-server/blob/main/docs/release-notes.asciidoc) file. [(Sample PR)](https://github.com/elastic/apm-server/pull/7956/files)
      * Patch version: Add a new section to existing release notes. ([Sample PR](https://github.com/elastic/apm-server/pull/8313/files))
    * Add `@elastic/obs-docs` as a reviewer.
```

Comment (endorama) [reply dev_docs/RELEASES.md:14]: Thanks for the suggestion. I'm confused about lines 14-21 though: we have 2 automation that perform what is described there and we are documenting them on line 18-22 of the new RELEASES.md. May you share the rationale for adding that part again?

Comment (bmorelli25) [reply dev_docs/RELEASES.md:14]: Looks like it would've helped if I had actually read the entire file 😵‍💫

Comment (bmorelli25) [reply dev_docs/RELEASES.md:14]: So the manual process has been replaced with an automated one.

> I think it is missing docs release process. It used to be done by docs team but it is now our responsibility to do so. Can you figure out at which stage that should happen and update the docs, and give an example?

@carsonip I'm not sure what's missing then

Comment (endorama) [reply dev_docs/RELEASES.md:14]: Thanks @bmorelli25! Then I will consider this conversation resolved (but won't do it in GitHub for future reference).

Comment (carsonip) [reply dev_docs/RELEASES.md:14]: Patch release automation does not include changelog changes. It has to be done in coordination with docs release manually. See https://github.com/elastic/dev/issues/2485 (private) and https://github.com/elastic/apm-server/pull/12680/files

File: dev_docs/RELEASES.md [hunk 7]
```diff
@@ -9,34 +9,17 @@ For patch releases, only the version on the existing major and minor version bra
...
-    Trigger a new beats update, once the beats branch is also created.
-    Remove the [changelogs/head.asciidoc](https://github.com/elastic/apm-server/blob/main/changelogs/head.asciidoc) file from the release branch.
-
-  * Main branch:
-    Update [.mergify.yml](https://github.com/elastic/apm-server/blob/main/.mergify.yml) with a new backport rule for the next version,
-    and update versions to next minor version, e.g. [#2804](https://github.com/elastic/apm-server/pull/2804).
-
-  The release manager will ping the teams, but you can already prepare this in advance on the day after Feature Freeze.
+* Trigger release workflow manually
+  * For **patch releases**: run the [`run-patch-release`](https://github.com/elastic/apm-server/actions/workflows/run-patch-release.yml) workflow.
```
Comment (carsonip) [inline dev_docs/RELEASES.md:19]: Describe what patch release automation does. It does not include changelog changes. It updates some versions iirc.

Comment (bmorelli25) [reply dev_docs/RELEASES.md:19]: For patch release notes we could add something like this:

* For patch releases, release notes must be manually added:
  * Add a new section to the existing release notes file ([Sample PR](https://github.com/elastic/apm-server/pull/12680)).
  * Review the [changelogs/head](https://github.com/elastic/apm-server/tree/main/changelogs/head.asciidoc) file and move all relevant release notes from `head.asciidoc` to `release_version.asciidoc`. If changes do not apply to the version being released, keep them in the `head.asciidoc` file.
  * Review the commits in the release to ensure all changes are reflected in the release notes.
  * Add your PR to the documentation release issue ([Sample Issue](https://github.com/elastic/dev/issues/2485)).

Comment (endorama) [reply dev_docs/RELEASES.md]: I added the suggested changes to [`837bc9a` (#12983)](https://github.com/elastic/apm-server/pull/12983/commits/837bc9ab1169fbc63d6e9b2665060d276f7a6a5e)

**Comment by endorama** [comment]
Thank you @kruskall for the super fast re-review! I'll wait @carsonip opinion before merging as we started this PR as a sort of brain dump from him and I'd like a final :heavy_check_mark: from him.

File: dev_docs/RELEASES.md (hunk 5 above)
Comment (endorama) [reply dev_docs/RELEASES.md:47]: Updated in [`43a62fe` (#12983)](https://github.com/elastic/apm-server/pull/12983/commits/43a62fea15407820201f21ed1d0db2a7246c386d)

**Review by carsonip — APPROVED** [review]

**Pull request merged**